### PR TITLE
fix: `RedisHandler::deleteMatching()` not deleting matching keys if cache prefix is used

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Supersedes and closes #8878 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by paulbalandan** [comment]
@samsonasik It seems the `AssertRegExpRector` is giving a wrong transformation of a `preg_match` assertion if the `$matches` variable is used in a subsequent assert.
See https://github.com/codeigniter4/CodeIgniter4/actions/runs/9449485549/job/26025852003?pr=8952

**Comment by samsonasik** [comment]
@paulbalandan could you reproduce at https://getrector.com/demo and report at https://github.com/rectorphp/rector/issues ? Thank you.

File: tests/system/Cache/Handlers/RedisHandlerTest.php [hunk 1]
```diff
@@ -129,44 +131,55 @@ public function testDelete(): void
...
     {
-        // Save 101 items to match on
-        for ($i = 1; $i <= 101; $i++) {
-            $this->handler->save('key_' . $i, 'value' . $i);
-        }
-
-        // check that there are 101 items is cache store
-        $dbInfo = explode(',', $this->handler->getCacheInfo()['db0']);
-        $this->assertSame('keys=101', $dbInfo[0]);
+        yield 'prefix' => ['key_1*', 13];
```
Comment (kenjis) [inline tests/system/Cache/Handlers/RedisHandlerTest.php:178]: Can you add the comment why the count is 13?

Comment (paulbalandan) [reply tests/system/Cache/Handlers/RedisHandlerTest.php:178]:
```suggestion
        // Given the key "key_1*", deleteMatching() should delete 13 keys:
        // key_1, key_10, key_11, key_12, key_13, key_14, key_15, key_16, key_17, key_18, key_19, key_100, key_101
        yield 'prefix' => ['key_1*', 13];
```

File: tests/system/Cache/Handlers/RedisHandlerTest.php [hunk 2]
```diff
@@ -129,44 +131,55 @@ public function testDelete(): void
...
 
-        // Checking that given the suffix "1", deleteMatching deletes 11 keys:
-        // (key_1, key_11, key_21, key_31, key_41, key_51, key_61, key_71, key_81, key_91, key_101)
-        $this->assertSame(11, $this->handler->deleteMatching('*1'));
+        yield 'suffix' => ['*1', 11];
```
Comment (paulbalandan) [inline tests/system/Cache/Handlers/RedisHandlerTest.php:180]:
```suggestion
        // Given the key "*1", deleteMatching() should delete 11 keys: 
        // key_1, key_11, key_21, key_31, key_41, key_51, key_61, key_71, key_81, key_91, key_101
        yield 'suffix' => ['*1', 11];
```

File: tests/system/Cache/Handlers/RedisHandlerTest.php [hunk 3]
```diff
@@ -129,44 +131,59 @@ public function testDelete(): void
...
-        $dbInfo = explode(',', $this->handler->getCacheInfo()['db0']);
-        $this->assertSame('keys=101', $dbInfo[0]);
+        // Given the key "key_1*", deleteMatching() should delete 13 keys:
+        // key_1, key_10, key_11, key_12, key_13, key_14, key_15, key_16, key_17, key_18, key_19, key_100, key_101
+        yield 'prefix' => ['key_1*', 13];
 
-        // Checking that given the suffix "1", deleteMatching deletes 11 keys:
-        // (key_1, key_11, key_21, key_31, key_41, key_51, key_61, key_71, key_81, key_91, key_101)
-        $this->assertSame(11, $this->handler->deleteMatching('*1'));
+        // Given the key "*1", deleteMatching() should delete 11 keys:
```
Comment (paulbalandan) [inline tests/system/Cache/Handlers/RedisHandlerTest.php:182]:
```suggestion
        // Given the key "*1", deleteMatching() should delete 11 keys:
```

**Review by kenjis — APPROVED** [review]

**Review by michalsn — APPROVED** [review]

**Pull request merged**